src/views/Simulation/components/TopBar/MapEntities.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import {
+  Checkbox,
+  MenuItem,
+  FormControl,
+  Select,
+  ListItemText,
+  InputLabel,
+  Chip,
+  Box,
+  Typography,
+  Switch,
+} from '@mui/material';
+import { useSimulationViewContext } from '../../SimulationViewContext';
+import { MAP_VIEW_FILTERS } from '../../constants/settings';
+
+export default function MapEntitiesDropdown() {
+  const { selectedMapFilters, setSelectedMapFilters, onlyCriticalPoints, setOnlyCriticalPoints } =
+    useSimulationViewContext();
+
+  const handleChange = (event) => {
+    const value = event.target.value;
+
+    if (value.includes('all')) {
+      setSelectedMapFilters(MAP_VIEW_FILTERS.map((e) => e.value));
+    } else {
+      setSelectedMapFilters(value);
+    }
+  };
+
+  const handleRemove = (value) => {
+    if (value === 'all') {
+      setSelectedMapFilters([]);
+    } else {
+      setSelectedMapFilters((prev) => prev.filter((v) => v !== value && v !== 'all'));
+    }
+  };
+
+  const renderValue = (selected) => {
+    if (selected.length === 0) return 'Entities';
+    if (selected.length === 1) {
+      const label = MAP_VIEW_FILTERS.find((opt) => opt.value === selected[0])?.label;
+      return label || 'Entities';
```

Comment:
When only one option is selected, could we return it in a Chip component too? Otherwise, the component looks like a "single option" dropdown list, it might be confusing for users 🙂